### PR TITLE
Adds RPN function LogicalAnd

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

LogicalAnd RPN function.

- [x] Testing code is based on https://github.com/tikv/tikv/pull/4574

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test.

### CODE REVIEW

File: src/coprocessor/dag/rpn_expr/impl_op.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+impl RpnFnLogicalAnd {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg0: &Option<i64>,
+        arg1: &Option<i64>,
+    ) -> Result<Option<i64>> {
+        // Intentionally not merging `None` and `Some(0)` conditions to be clear.
+        Ok(match arg0 {
```
Comment (AndreMouche) [inline src/coprocessor/dag/rpn_expr/impl_op.rs]: Maybe `match (arg0,arg1)` will be easier to review

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: Good idea!

File: src/coprocessor/dag/rpn_expr/impl_op.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        arg1: &Option<i64>,
+    ) -> Result<Option<i64>> {
+        // Intentionally not merging `None` and `Some(0)` conditions to be clear.
+        Ok(match (arg0, arg1) {
+            (None, Some(0)) => Some(0),
+            (None, _) => None,
+            (Some(0), _) => Some(0),
+            (Some(_), None) => None,
+            (Some(_), Some(0)) => Some(0),
+            (Some(_), Some(_)) => Some(1),
```
Comment (AndreMouche) [inline src/coprocessor/dag/rpn_expr/impl_op.rs]: Do you mean:
```
match (arg0, arg1) {
            (Some(0), _) | (_, Some(0)) => Some(0),
            (None, _) | (_, None) => None,
            _ => Some(1),
        }
```

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: Awesome!

**Pull request merged**